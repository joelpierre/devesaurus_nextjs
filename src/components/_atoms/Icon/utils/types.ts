import { ImgHTMLAttributes } from 'react';

export type TIconProps = ImgHTMLAttributes<HTMLImageElement> & {
  className?: string;
  inline?: boolean;
};

export enum EIconType {
  ChevronLeft = 'ChevronLeft',
  ChevronRight = 'ChevronRight',
  Logo = 'Logo',
  LogoDark = 'LogoDark',
  LogoCharacter = 'LogoCharacter',
  LogoCharacterDark = 'LogoCharacterDark',
  LogoSocial = 'LogoSocial',
  LogoSocialDark = 'LogoSocialDark',
  LogoSymbol = 'LogoSymbol',
  LogoSymbolDark = 'LogoSymbolDark',
  LogoText = 'LogoText',
  LogoTextDark = 'LogoTextDark'
}

export enum EIconTheme {
  Light = 'Light',
  Dark = 'Dark'
}
