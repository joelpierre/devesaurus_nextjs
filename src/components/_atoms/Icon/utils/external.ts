import { EIconType } from '@jpp/atoms/Icon/utils/types';

const IconUrls: Record<EIconType, string> = {
  [EIconType.ChevronLeft]: require('../../../../assets/svg/dev-chevron-left.svg'),
  [EIconType.ChevronRight]: require('../../../../assets/svg/dev-chevron-right.svg'),
  [EIconType.Logo]: require('../../../../assets/svg/logo.svg'),
  [EIconType.LogoDark]: require('../../../../assets/svg/logo-inv.svg'),
  [EIconType.LogoCharacter]: require('../../../../assets/svg/logo-character.svg'),
  [EIconType.LogoCharacterDark]: require('../../../../assets/svg/logo-character-inv.svg'),
  [EIconType.LogoSocial]: require('../../../../assets/svg/logo-social.svg'),
  [EIconType.LogoSocialDark]: require('../../../../assets/svg/logo-social-inv.svg'),
  [EIconType.LogoSymbol]: require('../../../../assets/svg/logo-symbol.svg'),
  [EIconType.LogoSymbolDark]: require('../../../../assets/svg/logo-symbol-inv.svg'),
  [EIconType.LogoText]: require('../../../../assets/svg/logo-text.svg'),
  [EIconType.LogoTextDark]: require('../../../../assets/svg/logo-text-inv.svg')
};

export default IconUrls;
