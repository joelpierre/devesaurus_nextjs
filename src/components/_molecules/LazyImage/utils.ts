export interface IRectangle {
  top: number;
  left: number;
  right: number;
  bottom: number;
}

export const isAllInViewport = (rect: IRectangle, viewportHeight: number, viewportWidth: number) => {
  return (
    rect.top >= 0 &&
    rect.left >= 0 &&
    rect.bottom <= viewportHeight &&
    rect.right <= viewportWidth
  );
};

export const isAnyInViewport = (rect: IRectangle, viewportHeight: number, viewportWidth: number) => {
  return (
    (rect.top >= 0 && rect.top <= viewportHeight ||
      rect.bottom >= 0 && rect.bottom <= viewportHeight) &&
    (rect.left >= 0 && rect.left <= viewportWidth ||
      rect.right >= 0 && rect.right <= viewportWidth)
  );
};

export const isElementInViewport = (element?: HTMLElement): boolean => {
  if (element === undefined
    || window === undefined
    || document === undefined
    || document.documentElement === null) {
    return false;
  }

  const rect = element.getBoundingClientRect();

  return isAllInViewport(
    rect,
    (window.innerHeight || document.documentElement.clientHeight),
    (window.innerWidth || document.documentElement.clientWidth)
  );
};

export const isElementNearViewport = (
  element: HTMLElement,
  portraitExpansionQuotient: number = 1,
  landscapeExpansionQuotient: number = 1): boolean => {

  if (element === undefined
    || window === undefined
    || document === undefined
    || document.documentElement === null) {
    return false;
  }

  const rect = element.getBoundingClientRect();

  let screenHeight = window.innerHeight || document.documentElement.clientHeight;
  let screenWidth = window.innerWidth || document.documentElement.clientWidth;

  const aspectRatio = screenWidth / screenHeight;

  const expansionQuotient = aspectRatio < 1
    ? portraitExpansionQuotient
    : landscapeExpansionQuotient;

  screenHeight += screenHeight * expansionQuotient;
  screenWidth += screenWidth * expansionQuotient;

  return isAnyInViewport(
    rect,
    screenHeight,
    screenWidth
  );
};

export const loadImageWithCanvas = (imageUrl: string): Promise<string> => {
  return new Promise<string>((resolve, reject) => {

    return loadImage(imageUrl).then(onLoadImage);

    // tslint:disable-next-line:only-arrow-functions
    function onLoadImage(image: HTMLImageElement) {
      const canvas = createImageCanvas(image);

      const context = canvas.getContext('2d');

      if (context !== null) {
        context.drawImage(image, 0, 0);
        const dataUrl = canvas.toDataURL('image/png');
        resolve(dataUrl);
      } else {
        reject('Unable to get 2d canvas.');
      }
    }
  });
};

export const loadImageWithXhr = (imageUrl: string): Promise<string> => {
  return new Promise<string>((resolve, reject) => {
    try {
      const xhr = new XMLHttpRequest();
      const blobHandler = () => {
        const url = window.URL.createObjectURL(xhr.response);
        resolve(url);
      };

      const svgHandler = () => {
        const svgData = encodeURIComponent(xhr.response);
        const url = `"data:image/svg+xml; charset=utf8, ${svgData}"`;
        resolve(url);
      };

      if (imageUrl.endsWith('.svg')) {
        xhr.onload = svgHandler;
      } else {
        xhr.onload = blobHandler;
        xhr.responseType = 'blob';
      }

      xhr.open('GET', imageUrl, true);
      xhr.send();

    } catch (error) {
      reject(error);
    }
  });
};

export const isFullBleed = (image: Core.IMedia) => {
  const getImageProp = (key: keyof Core.IMedia, defaultValue: number): number => {
    return image && image[key] as number || defaultValue;
  };

  const width = getImageProp('width', 0);
  const height = getImageProp('height', 0);
  const diff = Math.abs(width - height);

  // DIFF_TOLERANCE is an arbitrary about the diff is allowed to vary by.
  const DIFF_TOLERANCE = 5;
  return diff >= DIFF_TOLERANCE && width > 1000;
};

export const loadImage = (src: string): Promise<HTMLImageElement> => {
  return new Promise((resolve, reject) => {
    try {
      const image = new Image();
      image.crossOrigin = 'anonymous';
      image.src = src;
      image.onload = () => resolve(image);
      image.onerror = reject;
    } catch (error) {
      reject(error);
    }
  });
};

export const createImageCanvas = (image: HTMLImageElement): HTMLCanvasElement => {
  const canvas = document.createElement('canvas');
  const { naturalWidth, naturalHeight } = image;
  canvas.width = naturalWidth;
  canvas.height = naturalHeight;
  return canvas;
};
