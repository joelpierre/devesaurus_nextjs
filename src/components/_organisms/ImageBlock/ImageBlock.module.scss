.image-block {
  $root: &;
  position: relative;
  width: 100%;
}
