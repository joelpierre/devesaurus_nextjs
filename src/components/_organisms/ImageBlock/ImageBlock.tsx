import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

import { Section } from '@jpp/components/_shared/Grid/Section/Section';
import { Container } from '@jpp/components/_shared/Grid/Container/Container';
import { Flex } from '@jpp/components/_shared/Grid/Flex/Flex';
import { Row } from '@jpp/components/_shared/Grid/Row/Row';

import styles from './ImageBlock.scss';

type TImageBlock = Core.IAcfComponent;

export const ImageBlock: FunctionComponent<TImageBlock> = (
  {
    theme,
    page_theme,
    className
  }
) => {
  return (
    <Section
      data-test="component-"
      className={classNames([
        className,
        `theme--${theme ? theme : page_theme}`,
        styles.imageBlock
      ])}
    >
      <Container>
        <Row>
          <Flex className="flex">ImageBlock</Flex>
        </Row>
      </Container>
    </Section>
  );
};
