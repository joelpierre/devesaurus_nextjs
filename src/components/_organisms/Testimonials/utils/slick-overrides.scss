.testimonials-slider {
  .slick-list {
    overflow: hidden;
  }

  .slick-track {
    display: flex;
    align-items: center;
  }
}
