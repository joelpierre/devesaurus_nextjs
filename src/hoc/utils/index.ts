import { ComponentType } from 'react';

export const getDisplayName = <T extends object>(
  WrappedComponent: ComponentType<T>,
) => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
};
