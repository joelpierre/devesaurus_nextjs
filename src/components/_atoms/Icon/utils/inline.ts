import { FunctionComponent } from 'react';

import { EIconType, TIconProps } from '@jpp/atoms/Icon/utils/types';

const IconComponents: Record<EIconType, FunctionComponent<TIconProps>> = {
  [EIconType.ChevronLeft]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/dev-chevron-left.svg'),
  [EIconType.ChevronRight]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/dev-chevron-right.svg'),
  [EIconType.Logo]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo.svg'),
  [EIconType.LogoDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-inv.svg'),
  [EIconType.LogoCharacter]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-character.svg'),
  [EIconType.LogoCharacterDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-character-inv.svg'),
  [EIconType.LogoSocial]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-social.svg'),
  [EIconType.LogoSocialDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-social-inv.svg'),
  [EIconType.LogoSymbol]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-symbol.svg'),
  [EIconType.LogoSymbolDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-symbol-inv.svg'),
  [EIconType.LogoText]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-text.svg'),
  [EIconType.LogoTextDark]: require('-!svg-react-loader?props[]=class:svg!../../../../assets/svg/logo-text-inv.svg'),
};

export default IconComponents;
