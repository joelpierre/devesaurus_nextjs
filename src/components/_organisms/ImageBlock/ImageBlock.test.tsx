import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { findByTestAttr, matchSnapshot } from '../../../utils/test';

import ImageBlock from './ImageBlock';

const defaultProps = {
  module: {},
};

/**
 * Factory function to create a ShallowWrapper for the ImageBlock component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<ImageBlock {...setupProps} />);
};

describe('<ImageBlock/>', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the ImageBlock Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-');
    expect(component.length).toBe(1);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
