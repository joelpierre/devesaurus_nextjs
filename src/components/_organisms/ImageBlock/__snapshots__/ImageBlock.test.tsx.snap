// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageBlock/> should match snapshot 1`] = `
<Section
  className="theme--tint-alpha imageBlock"
  data-test="component-"
>
  <Container>
    <Row>
      <Flex
        className="flex"
      >
        ImageBlock
      </Flex>
    </Row>
  </Container>
</Section>
`;
